packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts

```diff
@@ -447,6 +435,44 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
       await this.updateObjectRelationships(input.id, input.update.isActive);
     }
 
+    if (input.update.labelIdentifierFieldMetadataId) {
+      const isSearchEnabled = await this.featureFlagService.isFeatureEnabled(
+        FeatureFlagKey.IsSearchEnabled,
+        workspaceId,
+      );
+
+      const isWorkspaceMigratedForSearch =
+        await this.featureFlagService.isFeatureEnabled(
+          FeatureFlagKey.IsWorkspaceMigratedForSearch,
+          workspaceId,
+        );
+
+      if (isSearchEnabled && isWorkspaceMigratedForSearch) {
```

Comment:
TODO: remove feature flag in next PR, shouldn't be necessary anymore

Reply:
done in this PR !